backend/app/services/dataset_item_service.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from datetime import datetime
+from io import BytesIO
+from typing import BinaryIO
+from uuid import UUID
+
+from app.schemas.dataset_item import DatasetItem
+from app.services.base import ResourceNotFoundError, ResourceType
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetItemService:
+    def create_dataset_item(self, file: BinaryIO) -> DatasetItem:
+        """Creates a new dataset item"""
+        raise NotImplementedError
+
+    def list_dataset_items(
+        self,
+        limit: int,  # noqa: ARG002
+        offset: int,  # noqa: ARG002
+        start_date: datetime | None,  # noqa: ARG002
+        end_date: datetime | None,  # noqa: ARG002
```

Comment:
```suggestion
        limit: int = 20,  # noqa: ARG002
        offset: int = 0,  # noqa: ARG002
        start_date: datetime | None = None,  # noqa: ARG002
        end_date: datetime | None = None,  # noqa: ARG002
```
Added reasonable default values